promql/engine.go

```diff
@@ -2049,10 +2113,18 @@ func (ev *evaluator) eval(ctx context.Context, expr parser.Expr) (parser.Value,
 		return mat, ws
 
 	case *parser.MatrixSelector:
+
+		vs := e.VectorSelector.(*parser.VectorSelector)
+		offset := durationMilliseconds(vs.Offset)
+		var mint, maxt int64
 		if ev.startTimestamp != ev.endTimestamp {
-			panic(errors.New("cannot do range evaluation of matrix selector"))
+			maxt = ev.endTimestamp - offset
+			mint = ev.startTimestamp - durationMilliseconds(e.Range) - offset
+		} else {
+			maxt = ev.startTimestamp - offset
+			mint = maxt - durationMilliseconds(e.Range)
```

Comment:
Do we need the `if ... else` case differentiation here? It looks like both cases calculate the exact same thing, just that the second case is a special case of the first one.

Reply:
you are right; simplified!